contracts/Unipool.sol

```diff
@@ -62,8 +72,26 @@ contract Unipool is LPTokenWrapper {
         _;
     }
 
-    constructor(IERC20 _uniswapTokenExchange) public {
+    constructor(IERC20 _uniswapTokenExchange, IUniswapV2Router01 _uniswapRouter, IERC20 _rewardToken) public {
         uniswapTokenExchange = _uniswapTokenExchange;
+        uniswapRouter = _uniswapRouter;
+        rewardToken = _rewardToken;
+
+        reinvestableToken = _getReinvestableToken();
+    }
+
+    function _getReinvestableToken() private view returns (IERC20) {
```

Comment:
Nitpicky but we always put private / internal functions after all of the external / public functions.